CMakeLists.txt

```diff
@@ -41,7 +41,7 @@ if(WITH_ASAN)
 endif()
 
 add_compile_options ( -Wall -Wextra -Winit-self -Wpointer-arith
-                      -Wcast-qual -Wcast-align )
+                      -Wcast-qual -Wcast-align -Wno-self-assign-overloaded)
```

Comment:
@igudich made the point that we should keep this warning. I agree, i'll remove the flag.